### PR TITLE
Fitting

### PR DESCRIPTION
**Pull request by zifn**
addresses issue #12 by introducing a fitting module that converts between spherical and cartesian coordinates and uses either MLE with gradient assent or binning with least squares fitting to get B_lm coefficients for the fit angular distributions.

### CODE REVIEW

File: src/fitting.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    assert M_xyz.shape[1] == 3
+    assert numb_bins > 0 and int(numb_bins) == numb_bins
+    M_sph = cart_to_spherical(M_xyz)
+
+    angular_hist, theta_hist, phi_hist = np.histogram2d(M_sph[:, 1], M_sph[:, 2], numb_bins,
+                                                                                    range=[[0, 2*np.pi], [0, np.pi]],
+                                                                                    density=True)
+    #estimate bin by center value
+    theta_hist = (theta_hist[1:] + theta_hist[:-1])/2
+    phi_hist = (phi_hist[1:] + phi_hist[:-1])/2
```
Comment (sajantanand) [inline src/fitting.py:155]: Why isn't the 0th element used in phi_hist and theta_hist?

File: src/fitting.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    # vectorize 2d histogram and input angles
+    angular_hist_flat = angular_hist.flatten()
+    theta_hist_flat = theta_hist.flatten()
+    phi_hist_flat = phi_hist.flatten()
+
+    # make ylm feature matrix
+    ylm_features, lm_order = Y_lm_features(theta_hist_flat, phi_hist_flat, L_max, only_even_Ls)
+
+    # solve for coeficients
+    B_lms = np.linalg.solve( ylm_features.T @ ylm_features, ylm_features.T @ angular_hist_flat)
```
Comment (sajantanand) [inline src/fitting.py:167]: Do we want to add a ridge regularizing parameter to force the Blm's to be smaller?

File: src/tests/test_fitting.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        for trial in range(1):
+            # make some random B_lms
+            sample_B_lms = []
+
+            for L in range(0, L_max + 1):
+                if  evens and L%2 == 1:
+                    continue
+                for _ in range(L+1):
+                    sample_B_lms.append(rng.normal())
+            sample_B_lms[0] = 3
```
Comment (sajantanand) [inline src/tests/test_fitting.py:58]: Why is B00 set to 3?

**Review by larchen — APPROVED** [review]
Looks good.

**Review by sajantanand — APPROVED** [review]
Looks good to me.

**Pull request merged**